src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -17,7 +17,7 @@ import scala.collection.JavaConversions._
 case class Group(
     id: PathId,
     apps: Map[AppDefinition.AppKey, AppDefinition] = defaultApps,
-    groups: Set[Group] = defaultGroups,
+    groupsById: Map[Group.GroupKey, Group] = defaultGroups,
```

Comment:
Both apps and groups are available via id. Shouldn't we have the same scheme? (rename apps to appsById and have  a method to get apps)